### PR TITLE
[1.x][LT] Add forward, backward test for linalg.gemm2

### PR DESCRIPTION
**Pull request by ChaiBapchya**
## Description ##
Related PR: https://github.com/apache/incubator-mxnet/pull/18718

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage:
- [x] Code is well-documented: 
- [x] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### CODE REVIEW

**Comment by ChaiBapchya** [comment]
@mxnet-label-bot add [pr-awaiting-review]

**Comment by ChaiBapchya** [comment]
Test results:
```
python -m pytest -s --exitfirst --verbose tests/nightly/test_large_array.py::test_gemm2
==================================================== test session starts =====================================================
platform linux -- Python 3.7.7, pytest-5.4.3, py-1.9.0, pluggy-0.13.1 -- /home/ubuntu/gemm2_test_venv/bin/python
cachedir: .pytest_cache
rootdir: /home/ubuntu/chai-mxnet
collected 1 item

tests/nightly/test_large_array.py::test_gemm2 [23:35:35] ../src/base.cc:84: Upgrade advisory: this mxnet has been built against cuDNN lib version 7501, which is older than the oldest version tested by CI (7600).  Set MXNET_CUDNN_LIB_CHECKING=0 to quiet this
warning.
PASSED

======================================================================================================================== warnings summary ========================================================================================================================
/home/ubuntu/gemm2_test_venv/lib/python3.7/site-packages/nose/importer.py:12
  /home/ubuntu/gemm2_test_venv/lib/python3.7/site-packages/nose/importer.py:12: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
    from imp import find_module, load_module, acquire_lock, release_lock

-- Docs: https://docs.pytest.org/en/latest/warnings.html
============================================================================================================ 1 passed, 1 warning in 192.18s (0:03:12) ============================================================================================================
```

File: tests/nightly/test_large_array.py [hunk 1]
```diff
@@ -1207,9 +1207,28 @@ def check_syrk_batch():
...
+
+        inp1=mx.nd.ones(shape=(SMALL_Y, LARGE_X))
+        inp1[0][0]=0.1
+        inp2=mx.nd.ones(shape=(LARGE_X, SMALL_Y))
+        inp1_grad, inp2_grad, out= run_gemm2(inp1,inp2)
+        assert out.asnumpy()[0][0] == LARGE_X
+        out.backward()
+        assert inp1_grad.asnumpy()[0][0] == SMALL_Y
+        assert_almost_equal(inp2_grad.asnumpy()[0][0],49.1)
+
```
Comment (access2rohit) [inline tests/nightly/test_large_array.py:1226]: output and gradient shape check ? I forgot to mention in other PRs :P
Need to ask others to fix it in their next PRs too.

Comment (ChaiBapchya) [reply tests/nightly/test_large_array.py:1228]: Added

**Comment by mseth10** [comment]
@ChaiBapchya file conflicts need resolving

**Comment by ChaiBapchya** [comment]
Reran after adding shape check
```
python3 -m pytest -s --exitfirst --verbose tests/nightly/test_large_array.py::test_linalg
========================================================= test session starts =========================================================
platform linux -- Python 3.7.7, pytest-5.4.1, py-1.8.1, pluggy-0.13.1 -- /home/ubuntu/anaconda3/bin/python3
cachedir: .pytest_cache
hypothesis profile 'default' -> database=DirectoryBasedExampleDatabase('/home/ubuntu/chai-mxnet/.hypothesis/examples')
rootdir: /home/ubuntu/chai-mxnet
plugins: remotedata-0.3.2, openfiles-0.4.0, astropy-header-0.1.2, hypothesis-5.8.3, arraydiff-0.3, doctestplus-0.5.0
collected 1 item

tests/nightly/test_large_array.py::test_linalg [23:12:33] ../src/base.cc:84: Upgrade advisory: this mxnet has been built against cuDNN
lib version 7501, which is older than the oldest version tested by CI (7600).  Set MXNET_CUDNN_LIB_CHECKING=0 to quiet this warning.
PASSED

========================================================== warnings summary ==========================================================$
/home/ubuntu/anaconda3/lib/python3.7/site-packages/nose/importer.py:12
  /home/ubuntu/anaconda3/lib/python3.7/site-packages/nose/importer.py:12: DeprecationWarning: the imp module is deprecated in favour o$
 importlib; see the module's documentation for alternative uses
    from imp import find_module, load_module, acquire_lock, release_lock

-- Docs: https://docs.pytest.org/en/latest/warnings.html
============================================== 1 passed, 1 warning in 142.52s (0:02:22) ==============================================
```

**Comment by ChaiBapchya** [comment]
@mxnet-label-bot update [pr-awaiting-merge]

**Pull request merged**